### ISSUE TITLE
Report an error if schema isn't loaded

### ISSUE THREAD

**Issue by ldodds** [theodi/csvlint#186]
If a user supplies a schema and we fail to load it, then the application should report an error so its clear that the schema hasn't been applied.

Currently it falls back to not applying any schema validation which is misleading.

As a first step, reporting a basic error would be an immediate improvement. 

Reporting why the schema is invalid would also be useful, although there are a range of possible errors:

* invalid JSON
* invalid regex
* invalid values for a constraint specification, e.g. a number or string where true/false is expected
* unknown constraint

The first two are definitely fatal errors. The last two are potentially just warnings. Depends if we want to work on a "best effort" basis.

**Comment by bcouston** [comment]
I'll be taking invalid regex

**Comment by quadrophobiac** [comment]
change error message for schema error, currently prints "Sorry, your CSV did not pass validation. Please review the errors and warnings below:"

**Comment by quadrophobiac** [comment]
lines 26 - 34 of Validation currently always eval false, would be worthwhile working out what condition they were intended to catch     

        if io.respond_to?(:tempfile)
          filename = io.original_filename
          csv = File.new(io.tempfile)
          io = File.new(io.tempfile)
        elsif io.class == Hash && !io[:body].nil?
          filename = io[:filename]
          csv_id = io[:csv_id]
          io = StringIO.new(io[:body])
        end

**Comment by Floppy** [comment]
@quadrophobiac you mean both of these conditions always fail? The first looks like it's intended to handle uploaded files, and the second ... not sure. Will look at the context....

**Comment by Floppy** [comment]
@quadrophobiac do you mean this section? https://github.com/theodi/csvlint/blob/master/app/models/validation.rb#L19

The line numbers seems different...

**Comment by quadrophobiac** [comment]
Yes apologies - I was referring to the line numbers on the error-reporting-schema_expanded-test-suite branch

**Comment by quadrophobiac** [comment]
I've confirmed non triggering by running the tests suite with a byebug in each conditional

**Comment by Floppy** [comment]
Ah, I looked on the fix-schema branch but not the other - my bad. Looking closer now.

**Comment by quadrophobiac** [comment]
rewrite or excising of package_controller  read_files method as it seems to be only triggered by the way a test functions rather than how user behaviour (in cucumber) would proceed

**Comment by bcouston** [comment]
Started a pull request on invalid regex functionality: https://github.com/theodi/csvlint.rb/pull/132

**Comment by pezholio** [comment]
theodi/csvlint.rb#132 is good to go - so we need to make sure the newly created errors are shown this end
